From 614f680dc67d98d16eb21a131cc43db24d08b758 Mon Sep 17 00:00:00 2001
From: Pino Toscano <ptoscano@redhat.com>
Date: Tue, 5 Dec 2017 15:36:34 +0100
Subject: [PATCH] lib: libvirt: stop using <shareable/> for appliance disk
 (RHBZ#1518517)

Commit aa9e0057b19e29f76c9a81f9aebeeb1cb5bf1fdb made the libvirt backend
use <shareable/> for the disk of the appliance, since at that time all
the instances were using the disk directly.
OTOH, commit 3ad44c866042919374e2d840502e53da2ed8aef0 switched to
overlays for read-only disks, including the appliance, so effectively
protecting the appliance.

Because of this, the libvirt backend does not need <shareable/> anymore.

Thanks to: Daniel Berrange, Richard W.M. Jones,  Peter Krempa.
---
 lib/launch-libvirt.c | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/lib/launch-libvirt.c b/lib/launch-libvirt.c
index e5121799e..5aae1b611 100644
--- a/lib/launch-libvirt.c
+++ b/lib/launch-libvirt.c
@@ -985,10 +985,6 @@ static int construct_libvirt_xml_appliance (guestfs_h *g, const struct libvirt_x
     }						\
   } while (0)
 
-/* <element/> */
-#define empty_element(element)					\
-  do { start_element(element) {} end_element (); } while (0)
-
 /* key=value attribute of the current element. */
 #define attribute(key,value)                                            \
   if (xmlTextWriterWriteAttribute (xo, BAD_CAST (key), BAD_CAST (value)) == -1){ \
@@ -1787,8 +1783,6 @@ construct_libvirt_xml_appliance (guestfs_h *g,
         == -1)
       return -1;
 
-    empty_element ("shareable");
-
   } end_element ();
 
   return 0;
-- 
2.13.2

